src/sbt-test/sbt-scalafix/relax-scalacOptions/test

```diff
@@ -1,22 +1,34 @@
-# we have 2 fatal warnings preventing compilation
+# check that regular compilation benefits from prior scalafix invocation
+# when the build does not contain any flag relaxed on scalafix invocation
+-> scalafixAll --check RemoveUnused
+-> checkLastCompilationCached
+> compile
+> checkLastCompilationCached
+
+# we now have 2 fatal warnings preventing compilation
+> set scalacOptions ++= Seq("-Xfatal-warnings")
 -> compile
 
 # ensure that a semantic rule can be ran despite warnings, to fix one of the warning
 > scalafixAll RemoveUnused
 
-# but that the zinc analysis file has not been persisted, otherwise the run with relax scalacOptions would
-# cause the next compile to be full (non-incremental) no matter if there was a previous successful compilation
--> checkZincAnalysisPresent
+# confirm that the rule was applied and that triggered compilation is cached once the updated file recompiled
+> scalafixAll --check RemoveUnused
+-> checkLastCompilationCached
+> scalafixAll --check RemoveUnused
+> checkLastCompilationCached
```

Comment:
this is the second goal of this PR: successive scalafix invocations on a cold environment (where no compile was run successfully) should benefit from incremental compilation